packages/bundle-source/src/index.js

```diff
@@ -309,7 +311,8 @@ function getExportWithNestedEvaluate(filePrefix) {
   ${computeExports}
 
   // Evaluate the entrypoint recursively, seeding the exports.
-  return computeExports(entrypoint, { require }, {});
+  const systemRequire = typeof require === 'undefined' ? undefined : require;
```

Comment:
Looks fine, but does this behave differently than just `const systemRequire = require;`? Is `typeof` exempt from some sort of referencing-missing-name warning or something?

Reply:
Yes, typeof is exempt from `ReferenceError`.

Reply:
Do note, though, that `ReferenceError` is not a "warning", it is a thrown exception.  `const systemRequire = require;` will throw `ReferenceError` if `require` is not declared.

You are not allowed to have undeclared variables in strict JS except in `typeof XXX`.  Sloppy mode allows you to assign to undeclared variables (which become globals), but IIUC that's the only exception (@erights will weigh in if I'm wrong).

Our SES `sloppyGlobals` mode gets this wrong because there's no way to completely emulate the sloppy behaviour using the 8 magic lines.